### PR TITLE
Add crashtracker

### PR DESCRIPTION
**Pull request by iamluc**
### Description

Need https://github.com/DataDog/libdatadog/pull/535

### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,7 +39,7 @@ RUN_TESTS_CMD := REPORT_EXIT_STATUS=1 TEST_PHP_SRCDIR=$(PROJECT_ROOT) USE_TRACKE
...
+RUST_FILES = $(BUILD_DIR)/Cargo.toml $(shell find components-rs -name '*.c' -o -name '*.rs' -o -name 'Cargo.toml' | awk '{ printf "$(BUILD_DIR)/%s\n", $$1 }' ) $(shell find libdatadog/{alloc,bin_tests,build-common,crashtracker,crashtracker-ffi,data-pipeline,data-pipeline-ffi,ddcommon,ddcommon-ffi,ddsketch,ddtelemetry,ddtelemetry-ffi,dynamic-configuration,ipc,profiling,profiling-ffi,profiling-replayer,remote-config,serverless,sidecar,sidecar-ffi,spawn_worker,symbolizer-ffi,tests,tools,trace-*,Cargo.toml} -type f \( -path "*/src*" -o -path "*/examples*" -o -path "*Cargo.toml" -o -path "*/build.rs" -o -path "*/tests/dataservice.rs" -o -path "*/tests/service_functional.rs" -o -path "*/benches/*.rs" \) -not -path "*/ipc/build.rs" -not -path "*/sidecar-ffi/build.rs")
```
Comment (bwoebi) [inline Makefile:42]: We don't use most of these files, right? We don't need any benches not profling, serverless or data-pipeline?
Please only specify what we actually need; these files determine what we ship in pecl release.

File: tests/ext/crashtracker_segfault.phpt [hunk 2]
```diff
@@ -0,0 +1,71 @@
+--TEST--
+Send crashtracker report when segmentation fault signal is raised and config enables it
+--SKIPIF--
+<?php
+if (!extension_loaded('posix')) die('skip: posix extension required');
+if (getenv('SKIP_ASAN') || getenv('USE_ZEND_ALLOC') === '0') die("skip: intentionally causes segfaults");
+if (getenv('PHP_PEAR_RUNTESTS') === '1') die("skip: pecl run-tests does not support XFAIL");
```
Comment (bwoebi) [inline tests/ext/crashtracker_segfault.phpt:7]: XFAIL? Is it because of `%A`?

File: tests/ext/crashtracker_segfault.phpt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+if (getenv('DD_TRACE_CLI_ENABLED') === '0') die("skip: tracer is disabled");
+if (file_exists("/etc/os-release") && preg_match("/alpine/i", file_get_contents("/etc/os-release"))) die("skip Unsupported LIBC");
```
Comment (bwoebi) [inline tests/ext/crashtracker_segfault.phpt:9]: Is this for this test or does alpine generally not support crashtracking?

File: tests/ext/crashtracker_segfault.phpt (hunk 2 above)
Comment (iamluc) [reply tests/ext/crashtracker_segfault.phpt]: yes, it's because of `%A`, I updated the comment

File: Makefile (hunk 1 above)
Comment (iamluc) [reply Makefile:42]: They were required to compile. But I finally found why.
Changing
```
datadog-crashtracker-ffi = { path = "../libdatadog/crashtracker-ffi", features = ["collector"] }
```
to
```
datadog-crashtracker-ffi = { path = "../libdatadog/crashtracker-ffi", default-features = false, features = ["collector"] }
```

Now, added crates are `crashtracker`, `crashtracer-ffi`, `dynamic-configuration` and `remote-config`

**Pull request merged**